mm2src/coins/eth.rs

```diff
@@ -1646,18 +1672,27 @@ impl EthCoin {
     }
 
     fn approve(&self, spender: Address, amount: U256) -> EthTxFut {
-        match &self.coin_type {
-            EthCoinType::Eth => panic!(),
-            EthCoinType::Erc20 {
-                platform: _,
-                token_addr,
-            } => {
-                let function = try_fus!(ERC20_CONTRACT.function("approve"));
-                let data = try_fus!(function.encode_input(&[Token::Address(spender), Token::Uint(amount),]));
+        let coin = self.clone();
+        let fut = async move {
+            let token_addr = match coin.coin_type {
+                EthCoinType::Eth => return ERR!("'approve' is expected to be call for ERC20 coins only"),
+                EthCoinType::Erc20 { token_addr, .. } => token_addr,
+            };
+            let function = try_s!(ERC20_CONTRACT.function("approve"));
+            let data = try_s!(function.encode_input(&[Token::Address(spender), Token::Uint(amount)]));
 
-                self.sign_and_send_transaction(0.into(), Action::Call(*token_addr), data, U256::from(APPROVE_GAS_LIMIT))
-            },
-        }
+            let gas_limit = try_s!(
```

Comment:
We shouldn't use `try_s!` in new code since it's deprecated

Reply:
You are right! Thanks)
Will remove it. Also, I'll ask you to re-review the PR again when I finish the following :)
> I would suggest using estimation gas instead of hardcoded gas limits for all EtomicSwap contract calls